polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -84,6 +85,34 @@ export function Item({
     </span>
   ) : null;
 
+  const matchState = matchStateForItem(
+    {url, matches, exactMatch, matchPaths, excludePaths},
+    location,
+  );
+
+  const matchingSubNavigationItems = subNavigationItems.filter((item) => {
+    const subMatchState = matchStateForItem(item, location);
+    return (
+      subMatchState === MatchState.MatchForced ||
+      subMatchState === MatchState.MatchUrl ||
+      subMatchState === MatchState.MatchPaths
+    );
+  });
+
+  const childIsActive = matchingSubNavigationItems.length > 0;
+
+  const selected =
+    selectedOverride == null
+      ? matchState === MatchState.MatchForced ||
+        matchState === MatchState.MatchUrl ||
+        matchState === MatchState.MatchPaths
+      : selectedOverride;
+
+  const icon =
+    polarisSummerEditions2023 && (selected || childIsActive)
+      ? matchedItemIcon || baseIcon
```

Comment:
Super minor nit for readability, since we're checking for null / undefined

```suggestion
      ? matchedItemIcon ?? baseIcon
```